virttest/qemu_storage.py

```diff
@@ -347,6 +357,48 @@ def get_image_repr(image, params, root_dir, representation=None):
     return func(image, params, root_dir)
 
 
+class _FilterConfigGatherer(object):
+    """ This class can be used to gather filter-specific configuration
+        from test params to generate fully-valid image_meta dicts.
+        The filter-specific image_meta should be obtained by calling
+        the gather_filter_config method.
+        This class should contain as much private methods as filters that
+        should be supported for image_meta generation. In other words,
+        each supported filter should contain a _<filter_type> implemented
+        in this class.
+    """
+
+    @classmethod
+    def gather(cls, filter_type, params):
+        """ Gathers filter-specific config from params and collects
+            them properly on an OrderedDict.
+            This method is in charge of calling the appropriate private
+            class method
+            :param filter_type: Filter type for which data will be gathered
+            :type filter_type: str
+            :param params: Collection of parameters from which data will be
+                           gathered
+            :type params: Ordered Dict
+            :return: Ordered Dict
+        """
+        # Calls the adequate function for us
+        try:
+            gather_func = getattr(cls, f"_{filter_type}")
+        except AttributeError:
+            # Filter is not supported
+            raise NotImplementedError("Image configuration meta generation "
+                                      "support hasn't been implemented for "
+                                      f"{filter_type} filter yet.")
+        return gather_func(filter_type, params)
+
+    @staticmethod
+    def _compress(filter_type, params):
+        """ Gives an empty OrderedDict (as compress filter doesn't have
+            specific attributes related to the filter)
+        """
+        return collections.OrderedDict()
```

Comment:
return collections.OrderedDict({'driver': 'compress'})

Reply:
This makes it much more clean and flexible, thanks!